addons/html_editor/static/src/main/list/list_plugin.js

```diff
@@ -42,54 +72,29 @@ export class ListPlugin extends Plugin {
         "dom",
         "color",
     ];
+    toolbarListSelectorKey = reactive({ value: 0 });
     resources = {
-        user_commands: [
-            {
-                id: "toggleList",
-                run: this.toggleListCommand.bind(this),
-            },
-            {
-                id: "toggleListUL",
-                title: _t("Bulleted list"),
-                description: _t("Create a simple bulleted list"),
-                icon: "fa-list-ul",
-                run: () => this.toggleListCommand({ mode: "UL" }),
-            },
-            {
-                id: "toggleListOL",
-                title: _t("Numbered list"),
-                description: _t("Create a list with numbering"),
-                icon: "fa-list-ol",
-                run: () => this.toggleListCommand({ mode: "OL" }),
-            },
-            {
-                id: "toggleListCL",
-                title: _t("Checklist"),
-                description: _t("Track tasks with a checklist"),
-                icon: "fa-check-square-o",
-                run: () => this.toggleListCommand({ mode: "CL" }),
-            },
-        ],
-        toolbar_groups: withSequence(30, { id: "list" }),
+        user_commands: Object.entries(listCommands).map(([id, command]) => ({
+            id,
+            ...omit(command, "mode"),
+            run: () => this.toggleListCommand({ mode: command.mode }),
+        })),
         toolbar_items: [
-            {
-                id: "bulleted_list",
-                groupId: "list",
-                commandId: "toggleListUL",
-                isActive: this.isListActive("UL"),
-            },
-            {
-                id: "numbered_list",
-                groupId: "list",
-                commandId: "toggleListOL",
-                isActive: this.isListActive("OL"),
-            },
-            {
-                id: "checklist",
-                groupId: "list",
-                commandId: "toggleListCL",
-                isActive: this.isListActive("CL"),
-            },
+            withSequence(5, {
+                id: "list",
+                groupId: "list_and_alignment",
+                title: _t("List"),
+                Component: ListSelector,
+                props: {
+                    items: listSelectorItems.map(({ id, commandId }) => ({
+                        id,
+                        ...pick(listCommands[commandId], "title", "icon", "mode"),
```

Comment:
IMO this is less readable than the duplication of before. However, I suspect it has been done because `title` used to be missing and we probably don't want to write that one twice as we will easily forget to update one when we update the other in the future.

Reply:
repushed!